backend/test/Test/API/DashboardSpec.hs

```diff
@@ -0,0 +1,263 @@
+      apiTry env (deAnalyseNewSubExp dClient unknownSqlId seReq) `shouldThrow`
+        errorWithStatus notFound404
+
+  describe "POST /subExperiment/{subExperimentId}/new" $ do
+    it "allows to create a new subExperiment" $ \env -> do
+      experiment <- apiTry env $ deNewSubExp dClient seId seReq
+      seCheckNew experiment
+
+    it "throws a 404 for unknown subExperiment" $ \env -> do
+      apiTry env (deNewSubExp dClient unknownSqlId seReq) `shouldThrow`
+        errorWithStatus notFound404
+
+  describe "DELETE /subExperiment/{subExperimentId}" $ do
+    it "allows to delete subExperiment" $ \env -> do
+      _ <- apiTry env $ deDeleteSubExp dClient seId
+      apiTry env (deGetSubExperiment dClient seId) `shouldThrow`
+        errorWithStatus notFound404
+
+    it "throws a 400 for primary subExperiment" $ \env -> do
+      apiTry env (deDeleteSubExp dClient sepId) `shouldThrow`
+        errorWithStatus badRequest400
+
+    it "throws a 404 for unknown subExperiment" $ \env -> do
+      apiTry env (deDeleteSubExp dClient unknownSqlId) `shouldThrow`
+        errorWithStatus notFound404
+  where
+    seId, sepId, senId :: SubExperimentId
+    seId = SqlId 19
+    sepId = SqlId 1
+    senId = SqlId 20
+
+    seSample :: SubExperimentResp
+    seSample = SubExperimentResp
+      { serName = "qwe"
+      , serIsSuspicious = False
+      , serResult = Right Params4PL
+        { p4plA = 29.028170157560027
+        , p4plB = -4.355374922258996
+        , p4plC = 53.2364098216494
+        , p4plD = 1000.4974345758565
+        }
+      }
+
+    seCheck :: (SubExperimentId, SubExperimentResp) -> SubExperiment -> IO ()
+    seCheck (sei, ser) (WithId id_ response) = do
+      id_ `shouldBe` sei
+      response `shouldBe` ser
+
+    seCheckLast, seCheckUpdated, seCheckSuspicious, seCheckPrimary, seCheckNew
+      :: SubExperiment -> IO ()
+    seCheckLast = seCheck (seId, seSample)
+    seCheckUpdated = seCheck (seId, seSample { serName = "rename" })
+    seCheckSuspicious = seCheck (seId, seSample { serIsSuspicious = True })
+    seCheckPrimary = seCheck (sepId, seSample { serName = "Primary" })
+    seCheckNew = seCheck (senId, seSample { serName = nserName seReq, serResult = aSample })
+
+    mId :: MeasurementId
+    mId = SqlId 1
+
+    mSample :: MeasurementResp
+    mSample = MeasurementResp
+      { mrConcentration = 10.0
+      , mrSignal = 1000.0
+      , mrIsEnabled = True
+      }
+
+    mCheckFirst :: Measurement -> IO ()
+    mCheckFirst (WithId id_ response) = do
+      id_ `shouldBe` mId
+      response `shouldBe` mSample
+
+    seReq :: NewSubExperimentReq
+    seReq = NewSubExperimentReq
+      { nserName = "newSubExperiment"
+      , nserChanges = one mId
+      }
+
+    aSample :: AnalysisResult
+    aSample = Right Params4PL
+      { p4plA = 29.142616408435742
+      , p4plB = -4.357823061054072
+      , p4plC = 53.239595653095996
+      , p4plD = 1000.3020103924559
```

Comment:
It's very fragile to hardcode these numbers because any minor change in Python code will cause these tests to break (AFAIU). I think it's better to call `analyse4PLOne` from test code and compare `AnalysisResult` received from API with the one received from `analyse4PLOne`.

Reply:
hm.. I do not like this too. Thanks for an advice, I'll do it

Reply:
I decided to not check exact values here, cause we already have such a tests, and here the main goal is check that response is correct, not computation results. I didn't want to duplicate some code from `Test.DashboardSpec`. 
What do you think?

Reply:
Okay, makes sense.